packages/storebuilder/src/contexts/StoreSetupProvider.tsx

```diff
@@ -1,120 +1,314 @@
 		});
 
+		console.log('submitForm: ', data);
+
 		handleActionRequest(data).then(() => {
 			removeEventListener('beforeunload', beforeUnloadListener);
-			const navigateTo = isNextLookAndFeel ? '#/wizard/look-and-feel' : '';
-			window.location.assign(`${ SITEBUILDER_URL }${ navigateTo }`);
+			window.location.assign(`${ STOREBUILDER_URL }`);
 		}).catch(() => handleError());
 	};
 
+	const setStateAndFormValue = (prop: keyof StoreSetupFormItemsInterface, value: string, touched?: boolean) => {
+		const formData = storeSetupState.form;
+		formData[ prop ].value = value;
+		formData[ prop ].touched = typeof touched === 'boolean' ? touched : true;
+
+		setStoreSetupState({
+			...storeSetupState,
+			[ prop ]: value,
+			form: formData
+		});
+	};
+
 	const setFormValue = (
-		prop: keyof FtcFormItemsInterface,
+		prop: keyof StoreSetupFormItemsInterface,
 		value: string,
 		touched: boolean = true
 	) => {
-		if (ftcState.form[ prop ].value === value) {
+		if (storeSetupState.form[ prop ].value === value) {
 			return;
 		}
 
-		const formData = ftcState.form;
+		const formData = storeSetupState.form;
 		formData[ prop ].value = value;
 		formData[ prop ].touched = touched;
-		setFtcState({
-			...ftcState,
+
+		setStoreSetupState({
+			...storeSetupState,
 			form: formData
 		});
 	};
 
-	const resetFormValue = (prop: keyof FtcFormItemsInterface) => {
-		const formData = ftcState.form;
-		formData[ prop ].value = '';
+	const resetFormValue = (prop: keyof StoreSetupFormItemsInterface) => {
+		const formData = storeSetupState.form;
+		formData[ prop ].value = Array.isArray(formData[ prop ].value) ? [] : '';
 		formData[ prop ].touched = false;
 
-		setFtcState({
-			...ftcState,
+		setStoreSetupState({
+			...storeSetupState,
 			form: formData
 		});
 	};
 
+	const isScreenTouched = (): boolean => {
+		const { form } = storeSetupState;
+		let fields: string[] = [];
+
+		switch (currentStep) {
+		case 1:
+			fields = locationScreenFormProps;
+			break;
+		case 2:
+			fields = storeDetailsScreenFormProps;
+			break;
+		}
+
+		const matches = fields.filter((property) => form[ property as keyof StoreSetupFormItemsInterface ].touched);
+
+		return matches.length > 0;
+	};
+
+	const resetFormValues = () => {
+		switch (currentStep) {
+		case 1:
+			return locationScreenFormProps.map((property) => resetFormValue(property as keyof StoreSetupFormItemsInterface));
+		case 2:
+			return storeDetailsScreenFormProps.map((property) => resetFormValue(property as keyof StoreSetupFormItemsInterface));
+		}
+	};
+
 	const setIsLoading = (isLoading: boolean) => {
-		setFtcState({
-			...ftcState,
+		setStoreSetupState({
+			...storeSetupState,
 			isLoading
 		});
 	};
 
+	const getRegions = () => {
+		return storeSetupState?.regions;
+	};
+
+	const setRegion = (region: string) => {
+		if (region === storeSetupState.form.region.value) {
+			return;
+		}
+
+		setIsLoading(true);
+		window.wp.apiRequest({ path: `/wc/v3/data/countries/${ region }` })
```

Comment:
I like this suggestion! On my todo list is to make a separate package in the monorepo that houses shared utilities, and this would fit nicely into that. 👍